### ISSUE TITLE
Cosmos: TimeOnly arithmetic is translated incorrectly

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#35311]
The following test:

```c#
[ConditionalTheory]
[MemberData(nameof(IsAsyncData))]
public virtual Task TimeOnly_subtract_TimeOnly(bool async)
    => AssertQuery(
        async,
        ss => ss.Set<BasicTypesEntity>().Where(b => b.TimeOnly - new TimeOnly(10, 0, 0) == new TimeSpan(0, 0, 15, 50, 500)));
```

... generates the following SQL:

```sql
SELECT VALUE c
FROM root c
WHERE ((c["TimeOnly"] - "10:00:00") = "00:15:50.5000000")
```

We should either correct the translation or block it. Check arithmetic over other temporal types to make sure we're doing the right thing.

**Comment by ChrisJollyAU** [comment]
@roji Was have a think on this. Don't really have a proper function to use. Closes would be `DateTimeDiff` but that you have to specify the date time part you want to operate on (e.g. hours, minutes etc) and not multiple parts at once.

What might be easier is to use `DateTimeToTicks`. Do this for both sides, do the subtract (or add etc) and if needed convert back to date time with `TicksToDateTime`

Thoughts?
Probably be able to sort out DateOnly, TimeOnly, maybe TimeSpan with this

**Comment by roji** [comment]
@ChrisJollyAU yeah, something along those lines might be possible; the question is if the Cosmos functions can deal with a time-only value as a string (`10:00:00`), e.g. combine it to a full timestamp, which can then be converted to ticks or whatever...

**Comment by ChrisJollyAU** [comment]
Going by the docs the input has to be in an ISO 8601 format https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/query/datetimetoticks so needs both date and time

Could probably set the date part to `0001-01-01`

**Comment by roji** [comment]
@ChrisJollyAU that would mean doing string concatenation, something like `DateTimeToTicks('0001-01-01T' + x.TimeProperty)`; definitely possible, but it goes beyond what we usually do... At a certain point, either a provider/database supports a type, or it does not - we generally try to avoid going to far in faking type support for databases which lack support...

**Comment by ChrisJollyAU** [comment]
Yeah, definitely could end up with something like `DateTimeToTicks(...) - DateTimeToTicks(...) = DateTimeToTicks(...)` Little bit convoluted and not too pretty
